classes/BadpwFailedLoginsDAO.php

```diff
@@ -59,14 +59,15 @@ public function deleteObject(BadpwFailedLogins $badpwObj) : bool {
 	 * @return BadpwFailedLogins object Object matching the username
 	 */
 	public function getByUsername(string $username) : ?BadpwFailedLogins {
+		$username = substr($username, 0, 255); // Avoid database error
 		// Verify if the username is an email
 		if (filter_var($username, FILTER_VALIDATE_EMAIL)) {
 			$user = Repo::user()->getByEmail($username);
 			if (!$user) {
 				return null;
 			}
 			$username = $user->getData('userName');
-		} elseif (strlen($username) > 32) { // Invalid username length
+		} elseif (strlen($username) > 255) { // Invalid username length
```

Comment:
the srtlen() seems unnecessary following the addition of line 62. could this elseif be a simple else?